### PR TITLE
Upgrade mimir-prometheus

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR upgrades mimir-prometheus to include the changes from https://github.com/grafana/mimir-prometheus/pull/609.

The description of that PR includes a summary of the changes, which I won't repeat here.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated.
- [n/a] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [n/a] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by jhesketh — APPROVED** [review]
I haven't verified the specific changes, but the vendoring lgtm

**Review by pracucci — APPROVED** [review]
I reviewed the mimir-prometheus PR and LGTM.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 * [BUGFIX] Query-frontend: Don't panic when using the `-query-frontend.downstream-url` flag. #7651
 * [BUGFIX] Ingester: when receiving multiple exemplars for a native histogram via remote write, sort them and only report an error if all are older than the latest exemplar as this could be a partial update. #7640
 * [BUGFIX] Ingester: don't retain blocks if they finish exactly on the boundary of the retention window. #7656
+* [BUGFIX] Ingester: correctly handle experimental native histograms when checkpointing WAL. #7744
```
Comment (aknuds1) [inline CHANGELOG.md:22]: Based on @bboreham's suggestion about a general mention of experimental native histograms bug fixes/improvements:
```suggestion
* [BUGFIX] Bug-fixes and improvements to experimental native histograms. #7744
```

**Review by aknuds1 — COMMENTED** [review]
Just suggesting a more general changelog entry about native histograms bug fixes/improvements, in line with what Bryan opined :)

**Pull request merged**